### PR TITLE
feat: complete rewrite, multi-platform, multi-architexture and more...

### PR DESCRIPTION
**Pull request by carlocorradini**
# Complete rewrite
1. Support for Windows, Mac and Linux 
1. Support for multiple architectures
1. Tests (WIP, help me to add more)
1. Typescript support
1. Better development experience with prettier, eslint, markdownlint, etc....
1. Pre-commit hooks
1. GitHub workflows
   Automatically test and build
   Automatically publish on NPM (Update `package.json` version, create a new Git tag and push)
   To work, you need to add GitHub secret (`GITHUB_TOKEN`) and NPM secret (`NPM_TOKEN`), please see `.github/workflows/ci.yml`
1. **And a lot more...**

**Add your thoughts if you have any ideas or something similar!**

### CODE REVIEW

**Comment by carlocorradini** [comment]
@gunar What do you think? 🥳

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          node-version: '16.x'
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Build Changelog
+        id: changelog
+        uses: mikepenz/release-changelog-builder-action@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (gunar) [inline .github/workflows/ci.yml:59]: Can we use the default, which is the ephemeral `env.GITHUB_TOKEN` token instead? Should have enough permissions to create a release.

Comment (carlocorradini) [reply .github/workflows/ci.yml:59]: `secrets.GITHUB_TOKEN` is already the default one.
See [_Using the GITHUB_TOKEN in a workflow_](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow) for more information.

**Comment by carlocorradini** [comment]
@gunar Before merging I'll update deps

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gunar) [reply .github/workflows/ci.yml:59]: Hmm I see. https://github.com/mikepenz/release-changelog-builder-action docs is off, it mentions `env.GITHUB_TOKEN`. You're right.

**Review by gunar — APPROVED** [review]
Happy to merge when you're ready.

**Comment by carlocorradini** [comment]
@gunar LGTM now
Ready to merge for me

**Comment by gunar** [comment]
Shouldn't the release action have run?

**Comment by carlocorradini** [comment]
@gunar I think you have to enable actions in the repo

**Comment by carlocorradini** [comment]
Note that to create a release you have to:
1. Edit version in package.json (already done for v2.0.0)
2. Create a tag (e.g. v.2.0.0) -> Same as version above
3. Push tag (git tag push origin v2.0.0)
4.

**Comment by gunar** [comment]
So the tag is not going to be created automatically? Why is that?

**Comment by carlocorradini** [comment]
Currently not.
It's better to keep this way to avoid an unwanted release. Only you or who have the privileges can create a tag and push, automatically creating a new release

**Comment by carlocorradini** [comment]
GitHub Actions not working is strange :/

**Comment by carlocorradini** [comment]
Have you tried to open ci.yml in GitHub Browser? Any error or warning?

**Comment by gunar** [comment]
It should be working.

image


For PRs, I might need to approve you first actually. Do you want to try and open a new PR?

image

**Comment by gunar** [comment]
Yes `ci.yml` opens for me.

**Comment by carlocorradini** [comment]
@mandric 
Glad you like it 🥳
Hope it's useful 😉

**Pull request merged**